subsys/bluetooth/host/att.c

```diff
@@ -2948,6 +2970,55 @@ static int bt_att_accept(struct bt_conn *conn, struct bt_l2cap_chan **ch)
 BT_L2CAP_CHANNEL_DEFINE(att_fixed_chan, BT_L2CAP_CID_ATT, bt_att_accept, NULL);
 
 #if defined(CONFIG_BT_EATT)
+static k_timeout_t credit_based_connection_delay(struct bt_conn *conn)
+{
+	/*
+	 * 5.3 Vol 3, Part G, Section 5.4 L2CAP COLLISION MITIGATION
+	 * ... In this situation, the Central may retry
+	 * immediately but the Peripheral shall wait a minimum of 100 ms before retrying;
+	 * on LE connections, the Peripheral shall wait at least 2 *
+	 * (connPeripheralLatency + 1) * connInterval if that is longer.
+	 */
+
```

Comment:
If this is in L2CAP why this is being implemented for EATT only? It would probably make more sense to implement the retry logic under L2CAP itself and spare ATT of these details.

Reply:
Hi Luiz,

This is EATT requirement, see https://github.com/zephyrproject-rtos/zephyr/issues/40009